### PR TITLE
Add json-schemas for Configuration json files

### PR DESCRIPTION
**Pull request by psyirius**
## Title
Add json-schemas for Configuration json files

## Type of Change
- [x] DX Enhancement
- [x] QOL

## Description
Add json-schemas for Configuration json files

## Impact
Will help with editing config json with intelligence and keeping a documentation of the expected json format.

## Future Plans
Validate it using some validators

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by psyirius** [comment]
@MyDrift-user Do you see this as any improvement to the DX? This Doesn't affect any functionality!

**Comment by MyDrift-user** [comment]
This seems to to be up-to date with current changes. It looks like it will break the special characters which's logic I have modified in my Refractor

File: Compile.ps1 [hunk 1]
```diff
@@ -65,7 +65,35 @@ Get-ChildItem "$workingdir\functions" -Recurse -File | ForEach-Object {
...
+        $firstLevelName = $firstLevelJsonList[$i]
+        if ($jsonAsObject.$firstLevelName.content -ne $null) {
+            $jsonAsObject.$firstLevelName.content = $jsonAsObject.$firstLevelName.content.replace('&','&#38;').replace('“','&#8220;').replace('”','&#8221;').replace("'",'&#39;').replace('<','&#60;').replace('>','&#62;').replace('—','&#8212;')
+            $jsonAsObject.$firstLevelName.content = $jsonAsObject.$firstLevelName.content.replace('&#39;&#39;',"&#39;") # resolves the Double Apostrophe caused by the first replace function in the main loop
+        }
+        if ($jsonAsObject.$firstLevelName.description -ne $null) {
+            $jsonAsObject.$firstLevelName.description = $jsonAsObject.$firstLevelName.description.replace('&','&#38;').replace('“','&#8220;').replace('”','&#8221;').replace("'",'&#39;').replace('<','&#60;').replace('>','&#62;').replace('—','&#8212;')
+            $jsonAsObject.$firstLevelName.description = $jsonAsObject.$firstLevelName.description.replace('&#39;&#39;',"&#39;") # resolves the Double Apostrophe caused by the first replace function in the main loop
+        }
+    }
```
Comment (og-mrk) [inline Compile.ps1:96]: @psyirius This code for handling special characters is not needed anymore, as we're dynamically generating the UI, therefore skipping the XAML Reader in the process, so I recommend removing it plus any comment related to it.

**Comment by ChrisTitusTech** [comment]
Lol, I was having fun with this implementation, but really the pester and unit tests in github actions it seems like overkill. I'm going to close it.

File: schemas/config/themes.json [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      "pattern": "^Transparent|(#[0-9A-Fa-f]{6})$"
+    },
+    "decimal": {
+      "type": "string",
+      "pattern": "^\\d+(\\.\\d+)?$"
+    }
+  },
+  "type": "object",
+  "patternProperties": {
+    "^[a-zA-Z_][a-zA-Z0-9_]*$": {
```
Comment (og-mrk) [inline schemas/config/themes.json:15]: @psyirius @ChrisTitusTech By selecting every theme entry, it'll make the tests always fail as now you don't need every single theming option in all theming entries, meaning the rule of having every theming option **required** should be set to the default theme.

more info can be found in [this PR](https://github.com/ChrisTitusTech/winutil/pull/2630)

**Pull request closed without merging**